libdrgn/debug_info.c

```diff
@@ -2433,6 +2433,21 @@ drgn_type_from_dwarf_internal(struct drgn_debug_info *dbinfo,
 					 "maximum DWARF type parsing depth exceeded");
 	}
 
+	/* If we get a declaration, try to follow it to a specification */
+	bool decl;
+	if (!dwarf_flag(die, DW_AT_declaration, &decl) && decl) {
```

Comment:
I think we should return `drgn_error_libdw()` if `dwarf_flag()` fails rather than ignoring it.